sql/wsrep_schema.cc

```diff
@@ -752,14 +753,16 @@ Wsrep_view Wsrep_schema::restore_view(THD* thd, const Wsrep_id& own_id) const {
     goto out;
   }
 
-  if (Wsrep_schema_impl::next_record(cluster_table) ||
+  if ((error= Wsrep_schema_impl::next_record(cluster_table)) != 0 ||
       Wsrep_schema_impl::scan(cluster_table, 0, cluster_uuid) ||
       Wsrep_schema_impl::scan(cluster_table, 1, view_id) ||
       Wsrep_schema_impl::scan(cluster_table, 2, view_seqno) ||
       Wsrep_schema_impl::scan(cluster_table, 3, proto_ver) ||
       Wsrep_schema_impl::scan(cluster_table, 4, capabilities)) {
-    end_cluster_scan= true;
-    goto out;
+    if (error != HA_ERR_END_OF_FILE) {
```

Comment:
May be instead of nested if , you can use error != HA_ERR_END_OF_FILE in &&

Reply:
Sachin, addressed your concern and got rid of nested `if()`